mobile/android/android-emulator-services.ts

```diff
@@ -400,20 +400,24 @@ class AndroidEmulatorServices implements Mobile.IAndroidEmulatorServices {
 		return avdInfo;
 	}
 
-	public getAvds(): string[] {
+	public getAvds(suggestedImage?:string): string[] {
 		let result: string[] = [];
 		if (this.$fs.exists(this.avdDir)) {
 			let entries = this.$fs.readDirectory(this.avdDir);
 			result = _.filter(entries, (e: string) => e.match(AndroidEmulatorServices.INI_FILES_MASK) !== null)
 				.map((e) => e.match(AndroidEmulatorServices.INI_FILES_MASK)[1]);
 		}
-		return result;
+		if (suggestedImage && !_.includes(result, suggestedImage)) {
```

Comment:
I'd suggest we simplify and refactor this a little bit. It took me a lot of time to understand the flow here.

Why not getAvds to return just the AVDs and then we can filter them out when we have to?